python/cudf/cudf/tests/test_groupby.py

```diff
@@ -2718,3 +2718,36 @@ def test_groupby_group_keys(group_keys, by):
     actual = g_group[["B", "C"]].apply(lambda x: x / x.sum())
     expected = p_group[["B", "C"]].apply(lambda x: x / x.sum())
     assert_eq(actual, expected)
+
+
+@pytest.fixture
+def df_ngroup():
+    df = cudf.DataFrame(
+        {
+            "a": [2, 2, 1, 1, 2, 3],
+            "b": [1, 2, 1, 2, 1, 2],
+            "c": ["a", "a", "b", "c", "d", "c"],
+        },
+        index=[1, 3, 5, 7, 4, 2],
+    )
+    df.index.name = "foo"
+    return df
+
+
+@pytest.mark.parametrize(
+    "by",
+    [
+        lambda: "a",
+        lambda: "b",
+        lambda: ["a", "b"],
+        lambda: "c",
+        lambda: pd.Series([1, 2, 1, 2, 1, 2]),
+        lambda: pd.Series(["x", "y", "y", "x", "z", "x"]),
+    ],
+)
+@pytest.mark.parametrize("ascending", [True, False])
+def test_groupby_ngroup(by, ascending, df_ngroup):
```

Comment:
These tests will fail on Pandas < 1.5, because it changed the behaviour around null groups in `groupby.ngroup`. Null groups have group number "-1" in Pandas < 1.5, and "nan/null" in Pandas >= 1.5.

Should we:

* Skip this test for Pandas < 1.5?
* Check for the Pandas version and change the `expect` data accordingly?

@galipremsagar, what would you suggest?

Reply:
If having a version-based comparison in pytest is trivial we could go with 2. or else we can go with 1(skip this pytest based on pandas version).

Reply:
I think it's fine to skip it we can't possibly verify this against each version of pandas and testing with current latest version of pandas should be good enough.

Reply:
Unfortunately it's not super trivial, as the behavioural differences are a bit more subtle than my initial comment. I'll go with the suggestion to skip it :)